### ISSUE TITLE
Using preview (ReactDnDMultiBackend -> MultiBackend?)

### ISSUE THREAD

**Issue by Jorundur** [LouisBrunner/react-dnd-multi-backend#1]
Hey, thanks for a great package!

I'm having some trouble understanding the preview, it would be great if there was some complete example of how this should be used besides just the one line
`<ReactDnDMultiBackend.Preview generator={this.generatePreview} />`

Where does ReactDnDMultiBackend come from? Shouldn't this be MultiBackend, i.e. the default export?

**Comment by LouisBrunner** [comment]
Hey, thanks for your interest! :)

I am going to add one soon then!

I am not sure I understand your second point. Are you talking about the package name, the Webpack library name or something else?

**Comment by Jorundur** [comment]
Sorry, my second point was badly worded (and I'm probably just not understanding something simple at the moment)...

I'm just thinking about `ReactDnDMultiBackend` (and `ReactDnDMultiBackend.Preview`) in the line `<ReactDnDMultiBackend.Preview generator={this.generatePreview} />`.

Where does that `ReactDnDMultiBackend` come from? Is it just from `require`-ing or `import`-ing the package? In which case it would have to be `import { MultiBackend as ReactDnDMultiBackend } from 'react-dnd-multi-backend';` instead of just `import MultiBackend from 'react-dnd-multi-backend';` since the default export in the package is called `MultiBackend`, right?

But maybe I'm just complicating things here... I'm just trying to use this preview thingy in mobile. 

So, here's my code simplified. I have a top-component called `QuestionPage`, in which I use MultiBackend:

```
import React, { Component } from 'react';
import Question from './Question';
import { DragDropContext } from 'react-dnd';
import MultiBackend from 'react-dnd-multi-backend';

class QuestionPage extends Component {
  render() {
    return (
      <div>
          <span>
            Question Page
          </span>
        </div>
        <Question />
      </div>
    );
  }
}

export default DragDropContext(MultiBackend)(QuestionPage);
```

Then in the `Question` component I have `Choice` components which are the draggable "cards". The Question and Choice components are basically the same as in react-dnd's sortable example: http://gaearon.github.io/react-dnd/examples-sortable-cancel-on-drop-outside.html (source is available).

Now, where in all this should `<ReactDnDMultiBackend.Preview generator={this.generatePreview} />` go?

---

PS: Just when writing this, I saw that `react-dnd-touch-backend` has some example with a drag preview, maybe I can use that as a reference.

**Comment by LouisBrunner** [comment]
I just added an example [here](https://github.com/LouisBrunner/react-dnd-multi-backend/blob/master/examples/index.html).
I don't use `import` or `require` in it, but `Preview` should just be a subset of the exported value, whatever its name. It should be `MultiBackend.Preview` in your case (not a 100% sure, I am not too familiar with node yet).

**Comment by LouisBrunner** [comment]
So, your issue made me work on this project again, and I finally managed to remove the last quirk of the project (having to force reload DragSources and DropTargets).
[Here is an example up-to-date](https://github.com/LouisBrunner/react-dnd-multi-backend/tree/master/src/examples).
You should be able to make your code work with
```js
import MultiBackend, { Preview } from 'react-dnd-multi-backend';
...
<Preview generator={this.generatePreview} />
...
export default DragDropContext(MultiBackend)(QuestionPage);
```

**Comment by Jorundur** [comment]
Thanks for a quick response, I'll try this!

**Comment by LouisBrunner** [comment]
Wow, sorry about that (I just saw it in my own app).
I published `2.1.8` which solve the problem (I should have used a more full-fledged app instead of the example to test the last version).

**Comment by Jorundur** [comment]
Thanks again, no error now 👍
